base/reinterpretarray.jl

```diff
@@ -0,0 +1,67 @@
+"""
+Gives a reinterpreted view (of element type T) of the underlying array (of element type S).
+If the size of `T` differs from the size of `S`, the array will be compressed/expanded in
+the first dimension.
+"""
+struct ReinterpretArray{T,S,N,A<:AbstractArray{S, N}} <: AbstractArray{T, N}
```

Comment:
Perhaps switch places for `S` and `N`? Typically for `AbstractArray`s the element type is the first parameter, and the rank the second parameter.